rust/toolchain.bzl

```diff
@@ -1,48 +1,48 @@
 """The rust_toolchain rule definition and implementation."""
 
-load(
-    "//rust/private:utils.bzl",
-    "find_cc_toolchain",
+load("//rust/private:utils.bzl", "dedent", "find_cc_toolchain")
+
+RustStdLibInfo = provider(
```

Comment:
Please move this provider definition into providers.bzl, and expose it through `rust_common` (https://github.com/bazelbuild/rules_rust/blob/2109a0a105baae04148bbe68b9e0f01996ea4e3b/rust/private/common.bzl#L28-L28)

Reply:
I'd name the provider as `StdlibInfo` (providers retain the location where they were declared in their identity, other providers named `StdlibInfo` will not conflict with this one because they will be declared in a different bzl file). If you prefer stricted camel casing I'm fine with that, but please also update the error message below.

Reply:
How's this look?